gt/main.c

```diff
@@ -293,15 +386,19 @@ decap_and_fill_eth(struct rte_mbuf *m,
 	 * the original packet to the destination.
 	 */
 	new_eth = rte_pktmbuf_mtod(m, struct ether_hdr *);
-	ether_addr_copy(&gt_conf->net->front.eth_addr,
-		&new_eth->s_addr);
 	/*
-	 * TODO The destination MAC address
-	 * comes from LLS block.
+	 * The destination MAC address comes from LLS block.
 	 */
+	eth_cache = gt_neigh_get_ether_cache(neigh,
+		pkt_info->inner_ip_ver, ip_dst, &gt_conf->net->front,
+		gt_conf->lcores[block_idx]);
+	if (eth_cache == NULL)
+		return -1;
+
+	if (unlikely(eth_cache->stale))
+		return -1;
```

Comment:
Just trying to sort out any differences between the ways that the cache entries are handled here vs. how they are handled in the GK block. Is there a reason why it is unlikely for a GT cache entry to be stale, whereas we don't have this check in the GK block?